src/Data/Repositories/ChannelRepository.cs

```diff
@@ -175,5 +179,97 @@ public async Task<(bool, string?)> SafeRemove(Channel type, bool forceClose = fa
+                listChannelsResponse = await client.Execute(x => x.ListChannelsAsync(new ListChannelsRequest(),
+                    new Metadata
+                    {
+                        {
+                            "macaroon", node.ChannelAdminMacaroon
+                        }
+                    }, null, default));
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Error while listing channels for node {NodeId}", node.Id);
+                return null;
+            }
+            
+            return listChannelsResponse;
+        }
+
+        public async Task<List<Channel>> GetAllManagedByUserNodes(string loggedUserId)
+        {
+            if (string.IsNullOrWhiteSpace(loggedUserId))
+                throw new ArgumentException("Value cannot be null or whitespace.", nameof(loggedUserId));
+            //Context
+            await using var applicationDbContext = await _dbContextFactory.CreateDbContextAsync();
+            
+            //Query
+            var channels = applicationDbContext.Channels
+            .Include(channel => channel.ChannelOperationRequests).ThenInclude(request => request.SourceNode).ThenInclude(x=> x.Users)
+            .Include(channel => channel.ChannelOperationRequests).ThenInclude(request => request.DestNode).ThenInclude(x=> x.Users)
+            .Include(channel => channel.ChannelOperationRequests).ThenInclude(request => request.Wallet)
+            .Include(channel => channel.ChannelOperationRequests).ThenInclude(request => request.ChannelOperationRequestPsbts)
+            .Include(x=> x.LiquidityRules)
+            .ThenInclude(x=> x.Node)
+            .Include(x=> x.LiquidityRules)
+            .ThenInclude(x=> x.Wallet).AsSplitQuery().ToList() //Go to memory
+            .Where(x => x.ChannelOperationRequests.Any(y => y.SourceNode.Users.Select(x => x.Id)
+                                                                .Contains(loggedUserId) ||
+                                                            y.DestNode.Users.Select(x => x.Id)
+                                                                .Contains(loggedUserId))).ToList();
+            
+            
+            return channels;
+            
+            
+
+            
+        }
+
+
+        public async Task<(bool, string?)> MarkAsClosed(Channel channel)
+        {
+            await using var applicationDbContext = await _dbContextFactory.CreateDbContextAsync();
+            
+            //check the channel exists in the node
+            var channelOperationRequest = channel.ChannelOperationRequests.FirstOrDefault();
+            
+            if (channelOperationRequest == null)
+            {
+                _logger.LogError("Error while marking as closed with id: {ChannelId}. No channel operation request found", channel.Id);
+                return (false, "No channel operation request found");
```

Comment:
This is a very Golang thing to do, why don't we use custom exceptions?

Reply:
I took it from Golang long time ago, true 👯 . I mean exceptions are cool but they need to be try-catched and sometimes they are a pain in performance. The main problem/feature is that all of the repositories have a golang-based error approach.